src/pint/models/timing_model.py

```diff
@@ -1502,6 +1496,57 @@ def delete_jump_and_flags(self, toa_table, jump_num):
             return
         self.components["PhaseJump"].setup()
 
+    def get_spin_freq(self, times, calctype="modelF0"):
+        """Return barycentric pulsar rotational frequency in Hz at specific times
+
+        Parameters
+        ----------
+        times : float or long double MJD (can be array), astropy.Time object, TOAs
+            The times (barycentric, inf freq) at which to compute the spin frequency
+        calctype : {'modelF0', 'numerical', 'taylor'}
```

Comment:
Perhaps a fourth option, that allowed passing in a fixed frequency, might be helpful to avoid fitter shenanigans where they realize they can rescale their errors by changing the F0 in the model? Or is that better handled by using the prefit model to do this computation?

Reply:
Where would that fixed frequency come from?  Would that be perhaps passed as an optional argument?  Although, I guess, as you are saying, it could default to being the prefit F0....

Reply:
My suggestion was that the caller be allowed to supply it, whether as an optional argument or actually as a floating-point number in `calctype`. But I'm not sure that's the right way to allow callers to avoid updating the frequency used mid-fit. Given the current implementation, Fitter instances could ensure that the model this was called on was the pre-fit model, not the model updated to reflect fitting.